Content.Server/Power/EntitySystems/ExtensionCableSystem.cs

```diff
@@ -258,14 +261,14 @@ private bool TryFindAvailableProvider(EntityUid owner, float range, [NotNullWhen
 
                 if (!provider.Connectable) continue;
 
-                if ((Transform(entity).LocalPosition - xform.LocalPosition).Length > Math.Min(range, provider.TransferRange)) continue;
+                var distance = (Transform(entity).LocalPosition - xform.LocalPosition).Length;
+                // Is the provider out of range or have we already found a closer one?
+                if (distance > Math.Min(range, provider.TransferRange) || distance >= closestDistanceFound) continue;
```

Comment:
Can Math.Min outside of the loop instead, also don't inline continues.

Reply:
Thanks! I've moved things accordingly